code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/ManualCarouselPushTemplate.kt

```diff
@@ -11,25 +11,28 @@
 
 package com.adobe.marketing.mobile.notificationbuilder.internal.templates
 
-import android.content.Intent
 import com.adobe.marketing.mobile.notificationbuilder.PushTemplateIntentConstants
 import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.NotificationData
 
-internal class ManualCarouselPushTemplate : CarouselPushTemplate {
+internal class ManualCarouselPushTemplate(data: NotificationData, intentAction: String?) : CarouselPushTemplate(data) {
```

Comment:
How do you feel about having `intentAction` as a field in the `IntentData` class that is automatically read in the `init` block? Subsequently, like Prashanth mentioned in a previous comment, we check the type of ` data is IntentData` and use `intentAction` in case of `IntentData` instead of accepting it here? 
It would help keep all the template class constructors uniform

Reply:
That will be perfect. Will try to edit IntentData to contain actions.
thanks for the suggestion